src/SignupWithMobile.js

```diff
@@ -0,0 +1,319 @@
+    paddingVertical: Platform.select({
+      android: 8,
+      ios: 11,
+    }),
+    borderBottomColor: "rgb(170, 170, 170)",
+    borderBottomWidth: 1,
+    fontSize: 16,
+  },
+  error: {
+    paddingVertical: 4,
+    color: "rgb(208, 2, 27)",
+    fontSize: 13,
+  },
+  submitButton: {
+    marginTop: 41,
+    borderColor: "rgb(151, 151, 151)",
+    borderRadius: 25,
+    height: 50,
+    alignItems: "center",
+    justifyContent: "center",
+    borderWidth: 1,
+  },
+  submitText: {
+    fontSize: 24,
+    color: "rgb(51, 51, 51)",
+  },
+  skipButton: {
+    marginTop: 42,
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  skipText: {
+    fontSize: 14,
+    color: "rgb(63, 68, 165)",
+  },
+});
+
+export type Mobile = {
+  countryCallingCode: string,
+  nationalNumber: string,
+};
+
+export type Props = {
+  style?: ViewStyle,
+  boxStyle?: ViewStyle,
+
+  headerTitle?: React.Node,
+  headerStyle?: TextStyle,
+  description?: React.Node,
+  descriptionStyle?: TextStyle,
+  title?: React.Node,
+  titleStyle?: TextStyle,
+
+  countryPickerProps?: CountryPickerProps,
+  mobileNumberProps?: TextInputProps,
+
+  submitButtonStyle?: ViewStyle,
+  submitButtonText?: React.Node,
+  submitButtonTextStyle?: TextStyle,
+
+  skipButtonStyle?: ViewStyle,
+  skipButtonText?: React.Node,
+  skipButtonTextStyle?: TextStyle,
+
+  loading?: boolean,
+
+  onPressSubmitButton?: (isValid: boolean, mobile: Mobile) => void,
+  onPressSkipButton?: () => void,
+};
+
+type State = {
+  countryCallingCode: string,
+  nationalNumber: string,
+  marginTop: number,
+};
+
+export default class SignupWithMobile extends React.PureComponent<
+  Props,
+  State
+> {
+  state = {
+    countryCallingCode: "",
+    nationalNumber: "",
+    marginTop: 0,
+  };
+
+  onChangeText = (text: string) => {
+    if (/^\d*$/.test(text)) {
+      this.setState({
+        nationalNumber: text,
+      });
+    }
+  };
+
+  onCountryCodeChange = (countryCallingCode: string) => {
+    this.setState({ countryCallingCode });
+  };
+
+  onPressSubmitButton = () => {
+    const { countryCallingCode, nationalNumber } = this.state;
+    try {
+      const phoneNumber = parsePhoneNumber(
+        `${countryCallingCode} ${nationalNumber}`
+      );
+      if (this.props.onPressSubmitButton) {
+        this.props.onPressSubmitButton(
+          phoneNumber ? phoneNumber.isValid() : false,
+          {
+            countryCallingCode,
+            nationalNumber,
+          }
+        );
+      }
+    } catch (e) {
+      if (this.props.onPressSubmitButton) {
+        this.props.onPressSubmitButton(false, {
+          countryCallingCode,
+          nationalNumber,
+        });
+      }
+    }
+  };
+
+  // The box should be middle of the screen, so we need to raise this component.
+  adjustLayout = (event: LayoutEvent) => {
+    // If currect height less than iPhone SE height, then don't adjust layout.
+    this.setState({
+      marginTop:
```

Comment:
If you apply this negative margin, will content be clipped in android?

Reply:
Android display fine.